src/components/CommentInput.jsx

```diff
@@ -0,0 +1,63 @@
+import { useState } from "react";
+import { ClipLoader } from "react-spinners";
+import { TiTick } from "react-icons/ti";
+import { RxCrossCircled } from "react-icons/rx";
+import * as api from "../utils/api";
+
+const CommentInput = ({ articleId, user, setComments }) => {
+    const [newComment, setNewComment] = useState("");
+    const [validComment, setValidComment] = useState(false);
+    const [showErrorMessage, setShowErrorMessage] = useState(false);
+    const [postingComment, setPostingComment] = useState(false);
+    const [commentPosted, setCommentPosted] = useState(false);
+    const { username } = user;
+
+    const isValidComment = (targetValue) => {
+        if (/\w|\d/g.test(targetValue)) {
+            setValidComment(true);
+        } else {
+            setValidComment(false);
+            setCommentPosted(false);
+            setShowErrorMessage(true);
+        }
+    }
+
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        if (validComment) {
+            setPostingComment(true);
+            api.postCommentByArticleId(articleId, username, newComment).then((data) => {
+                setPostingComment(false);
+                setCommentPosted(true);
+                setComments((currComments) => {
+                    return [data.comment, ...currComments];
+                })
+                setNewComment("");
+                });
```

Comment:
I'd recommend having a catch situation for your API calls so that errors can be seen if happening!